java-checks/src/test/files/checks/CORSCheck.java

```diff
@@ -1,53 +1,86 @@
     resp.setHeader("Content-Type", "text/plain; charset=utf-8");
-    resp.setHeader("Access-Control-Allow-Origin", "http://localhost:8080"); // Noncompliant [[sc=5;ec=19]]
-    resp.setHeader("Access-Control-Allow-Credentials", "true"); // Noncompliant [[sc=5;ec=19]]
-    resp.setHeader("Access-Control-Allow-Methods", "GET"); // Noncompliant
+    resp.setHeader("Access-Control-Allow-Origin", "*"); // Noncompliant [[sc=5;ec=19]]
     // header names are case insensitive. see https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive/5259004#5259004
-    resp.setHeader("access-control-allow-Methods", "GET"); // Noncompliant
+    resp.setHeader("Access-control-allow-Origin", "*"); // Noncompliant [[sc=5;ec=19]]
+    resp.setHeader("Access-Control-Allow-Origin", "http://localhost:8080"); // Compliant
+
+    resp.setHeader("Access-Control-Allow-Credentials", "true"); // Compliant
+    resp.setHeader("Access-Control-Allow-Credentials", "*"); // Compliant
+    resp.setHeader("access-control-allow-Methods", "GET"); // Compliant
+    resp.setHeader("Access-Control-Allow-Methods", "*"); // Compliant
 
     resp.addHeader("Content-Type", "text/plain; charset=utf-8");
-    resp.addHeader("Access-Control-Allow-Origin", "http://localhost:8080"); // Noncompliant [[sc=5;ec=19]]
-    resp.addHeader("Access-Control-Allow-Credentials", "true"); // Noncompliant
-    resp.addHeader("Access-Control-Allow-Methods", "GET"); // Noncompliant
-    resp.addHeader("Access-Control-Allow-Methods", null); // Noncompliant
-    resp.addHeader(null, "GET");
+    resp.addHeader("Access-Control-Allow-Origin", "*"); // Noncompliant [[sc=5;ec=19]]
+    resp.addHeader("Access-Control-Allow-Origin", "http://localhost:8080"); // // Compliant
+    resp.addHeader("Access-Control-Allow-Credentials", "true"); // Compliant
+    resp.addHeader("Access-Control-Allow-Methods", "GET"); // Compliant
+
+    resp.addHeader("Access-control-allow-Origin", null); // Compliant
+    resp.addHeader(null, "*"); // Compliant
 
     resp.getWriter().write("response");
   }
   // === Spring MVC Controller annotation ===
-  @CrossOrigin(origins = "http://domain1.com") // Noncompliant [[sc=4;ec=15]] {{Make sure that enabling CORS is safe here.}}
+  @CrossOrigin(origins = "*") // Noncompliant [[sc=4;ec=15]] {{Make sure that enabling CORS is safe here.}}
   @RequestMapping("")
   public class TestController {
     public String home(ModelMap model) {
       model.addAttribute("message", "ok ");
       return "view";
     }
 
-    @CrossOrigin(origins = "http://domain2.com") // Noncompliant
+    @CrossOrigin("*") // Noncompliant
     @RequestMapping(value = "/test1")
     public ResponseEntity<String> test1() {
       return ResponseEntity.ok().body("ok");
     }
+
+    @CrossOrigin // Noncompliant
+    @RequestMapping(value = "/test2")
+    public ResponseEntity<String> test2() {
+      return ResponseEntity.ok().body("ok");
+    }
+
+    @CrossOrigin("http://domain2.com") // Noncompliant
```

Comment:
I juste realized that the key `value` is an alias for `origins`, this one, and a few following ones are in fact compliant!